### PR TITLE
Use crosssbean_channel.

### PR DESCRIPTION
**Pull request by kaikai1024**

### CODE REVIEW

**Review by u2 — DISMISSED** [review]

**Review by boundless-forest — DISMISSED** [review]

File: pubsub_zeromq/src/lib.rs [hunk 1]
```diff
@@ -17,9 +17,10 @@
 
 #[macro_use]
 extern crate logger;
+extern crate crossbeam_channel as channel;
 extern crate zmq;
-use std::sync::mpsc::Receiver;
-use std::sync::mpsc::Sender;
+use channel::Receiver;
```
Comment (chaoticlonghair) [inline pubsub_zeromq/src/lib.rs:22]: Export `Sender` and `Receiver` is better: `pub extern crate ...` or `pub use channel:: ...`.

**Review by chaoticlonghair — CHANGES_REQUESTED** [review]

File: pubsub_zeromq/src/lib.rs (hunk 1 above)
Comment (kaikai1024) [reply pubsub_zeromq/src/lib.rs:22]: https://github.com/cryptape/cita-common/blob/ee1ee917b7953d55eadf3f8c102263473b493463/pubsub/src/lib.rs#L17

Check it.

Comment (kaikai1024) [reply pubsub_zeromq/src/lib.rs:22]: > Export `Sender` and `Receiver` is better: `pub extern crate ...` or `pub use channel:: ...`.

Other crates should  specify the rabbitmq/zeromq/kafka feature.

**Review by chaoticlonghair — APPROVED** [review]

File: pubsub_zeromq/src/lib.rs (hunk 1 above)
Comment (chaoticlonghair) [reply pubsub_zeromq/src/lib.rs:22]: Your solution is OK.
But, if I do this task, I will export `channel` in rabbitmq / zeromq / kafka, the pubsub import `channel` from rabbitmq / zeromq / kafka and export it.

Comment (u2) [reply pubsub_zeromq/src/lib.rs:22]: Anyone is ok for me.

**Review by u2 — APPROVED** [review]

File: pubsub_zeromq/src/lib.rs (hunk 1 above)
Comment (chaoticlonghair) [reply pubsub_zeromq/src/lib.rs:22]: > Anyone is ok for me.

The key different is the channels can be different in different implementations in my solution.
If users just use the core API of `channel`, they can not sense that.

In @kaikai1024 's solution, all implementations has a same `channel`, and if change it in any implementations, we have to change all `channel` in all implementations.
Suppose there is a situation that we have to benchmark two implementations of `channel`, we can just compare them for just one MQ implementation, but we have to update all code for all MQ implementations (for compile).

Comment (kaikai1024) [reply pubsub_zeromq/src/lib.rs:22]: 我看到 pubsub crate 里面封装了 `start_pubsub`，我理解就是屏蔽了后面的三种类型，使用户对这些不用关系，直接调用相同的接口就行了。如此，在逻辑上（已经实现的逻辑合不合适另外讨论）看，其实是 pubsub 后面隐藏了后三种 crate，我理解别的 crate 使用 pubsub 功能的话只引用 `pubsub` 这个 crate 就行了。如果这样，我觉得我这种是 ok 的，应为 使用者 channel 要和 `pubsub` 一样就行了。

你说的这种情况是实现冲突吧，即使这样，也可以单独跳过 pubsub crate 直接单独引用相关的 crate 也可以吧

Comment (kaikai1024) [reply pubsub_zeromq/src/lib.rs:22]: > Your solution is OK.
> But, if I do this task, I will export `channel` in rabbitmq / zeromq / kafka, the pubsub import `channel` from rabbitmq / zeromq / kafka and export it.

Get your point, another pr will solve this.

Comment (u2) [reply pubsub_zeromq/src/lib.rs:22]: 其实博宇的更好，不过我觉得都可以。开开这样，虽然会有写冗余，但是这样就是强制三个库使用同样的crossbean-channel，副作用。

**Pull request merged**